staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -372,6 +379,63 @@ func readAndCloseResponseBody(resp *http.Response) {
 	}
 }
 
+type ResponseCloser struct {
+	once     sync.Once
+	resp     *http.Response
+	respCopy io.ReadWriteCloser
+	err      error
+}
+
+func NewResponseCloser(resp *http.Response) *ResponseCloser {
+	return &ResponseCloser{
+		resp:     resp,
+		respCopy: &buffer{b: &bytes.Buffer{}},
```

Comment:
could defer allocation of this to when we actually need it in the `if r.resp.ContentLength <= maxBodySlurpSize` block